### DISCUSSION TITLE
API Search Params and relational data

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [atrocore/atrocore#282] [Q&A] [answered]
I'm using the API extensively and struggle with a few things that I don't understand - relational data:

I want to query all Products in a Category and get as much information about the categories as well. I check out the Network Tab and can go with a query like this:[1]

`/Product?where[0][type]=bool&where[0][value][]=onlyActive&where[0][value][]=linkedWithCategory&where[0][data][linkedWithCategory]=600596814bb260feb`

This returns the Products, but in order to get information about the Category I have to include `select=name,categoriesIds` and now I need to write out all the fields I want to query (all Product fields).

With requesting the `categoriesIds` I'm getting an array of IDs and an object with the `categoryNames` - but I can't get the translated categories Names.

I dug into the code (https://github.com/atrocore/atropim/blob/3d362130a9/app/SelectManagers/Product.php#L947) and noticed that categories are treated differently but I couldn't find where `categoriesIds` are coming from.

And I couldn't make sense of how to return more than `categoryName` especially since the plural version is nor really existent.
The Espo Docs https://docs.espocrm.com/development/api-search-params/ helped a little bit, but beyond simple queries, this is still a black box and needs some docs (I'm happy to share stuff that I found out here in this post).

A similar problem was described here: https://forum.espocrm.com/forum/developer-help/32381-get-all-linked-objects-with-1-api-call but I know that AtroCore is too far from the Espo source, so you probably to things different now.

**TL;DR:** I want to avoid sending one query over a list of products and then looping over each item and send a query to retrieve detailed Category information.  How to do that?

[1]: I used `category/:id/products` before - but I need all Products nested under a ParentCategory.

**Comment by atrolex** [comment, marked as answer]
With requesting the categoriesIds I'm getting an array of IDs and an object with the categoryNames - but I can't get the translated categories Names. - This is how it works now. Only CategoryName for main language is currently provided. To extend it is too complicated.  To get more data for categories you need to query categories.
Alternatively you can create your own module, which would create an access point, which will return the data you need.

**Reply by atrolex** [reply]
[1]: I used category/:id/products before - but I need all Products nested under a ParentCategory. - it is not possible with one single query, unless you provide all you categories of that category tree as a filter.
